test/quantum_objects.jl

```diff
@@ -16,19 +22,17 @@
     N = 10
     a = rand(ComplexF64, 10)
     # @test_logs (:warn, "The norm of the input data is not one.") QuantumObject(a)
-    @test_throws DomainError Qobj(a, type = Bra)
-    @test_throws DomainError Qobj(a, type = Operator)
-    @test_throws DomainError Qobj(a, type = SuperOperator)
-    @test_throws DomainError Qobj(a, type = OperatorBra)
-    @test_throws DomainError Qobj(a', type = Ket)
+    @test_throws ArgumentError Qobj(a, type = Operator)
+    @test_throws ArgumentError Qobj(a, type = SuperOperator)
+    @test_throws ArgumentError Qobj(a', type = Ket)
     @test_throws DomainError Qobj(a', type = Operator)
     @test_throws DomainError Qobj(a', type = SuperOperator)
-    @test_throws DomainError Qobj(a', type = OperatorKet)
+    @test_throws ArgumentError Qobj(a', type = OperatorKet)
     @test_throws DimensionMismatch Qobj(a, dims = [2])
-    @test_throws DimensionMismatch Qobj(a', dims = [2])
-    a2 = Qobj(a')
+    @test_throws DomainError Qobj(a', dims = [2])
+    a2 = Qobj(a', type = Bra)
     a3 = Qobj(a)
-    @test dag(a3) == a2
```

Comment:
Is this the only place we test synonym `dag` ?

Reply:
Yes, I think so.
Maybe we need to add another test for this synonym `dag`, or just keep `dag` here.

Reply:
Ok I replace it with dag